File: backend/src/main/java/com/back/domain/pet/entity/PetStatus.java
package com.back.domain.pet.entity;


import com.back.domain.pet.enums.PetStatusType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.*;

@Entity
@Getter
@Setter
@Table(name = "pet_status")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class PetStatus {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "pet_status_id")
    private Long id;

    @Column(name = "pet_status_type", nullable = false)
    @Enumerated(EnumType.STRING)
    private PetStatusType status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pet_id", nullable = false)
    private Pet pet;

    @Builder
    public PetStatus(PetStatusType status, Pet pet) {
        this.status = status;
        this.pet = pet;
    }

}

File: backend/src/main/java/com/back/domain/pet/entity/PetStatus.kt
package com.back.domain.pet.entity

import com.back.domain.pet.enums.PetStatusType
import jakarta.persistence.*

@Entity
@Table(name = "pet_status")
class PetStatus protected constructor(
    @Enumerated(EnumType.STRING)
    @Column(name = "pet_status_type", nullable = false)
    var status: PetStatusType,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pet_id", nullable = false)
    var pet: Pet
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "pet_status_id")
    val id: Long? = null


    fun updateStatus(newStatus: PetStatusType) {
        this.status = newStatus
    }

    companion object {
        fun create(status: PetStatusType, pet: Pet): PetStatus {
            return PetStatus(status, pet)
        }
    }

}
